pennylane/qchem/vibrational/pes_generator.py

```diff
@@ -0,0 +1,676 @@
+    return pes_twobody, None # pragma: no cover
+
+
+def _local_pes_twomode(
+    comm,
+    molecule,
+    scf_result,
+    freqs_au,
+    displ_vecs,
+    gauss_grid,
+    pes_onebody,
+    dipole_onebody,
+    method="rhf",
+    do_dipole=False,
+):
+    r"""Computes the two-mode potential energy surface on a grid in real space, along the normal coordinate directions
+    (or any directions set by the displ_vecs) for each processor.
+    Simultaneously, can compute the dipole two-mode elements."""
+
+    size = comm.Get_size()
+    rank = comm.Get_rank()
+    freqs = freqs_au * AU_TO_CM
+    init_geom = molecule.coordinates * BOHR_TO_ANG
+    nmodes = len(freqs)
+
+    all_mode_combos = [(mode_a, mode_b) for mode_a in range(nmodes) for mode_b in range(mode_a)]
+
+    all_jobs = [
+        [i, pt1, j, pt2]
+        for (i, pt1), (j, pt2) in itertools.product(enumerate(gauss_grid), repeat=2)
+    ]
+    jobs_on_rank = np.array_split(all_jobs, size)[rank]
+
+    local_pes_twobody = np.zeros((len(all_mode_combos) * len(jobs_on_rank)))
+
+    if do_dipole:
+        local_dipole_twobody = np.zeros((len(all_mode_combos) * len(jobs_on_rank), 3), dtype=float)
+        ref_dipole = get_dipole(scf_result, method)
+
+    for mode_idx, [mode_a, mode_b] in enumerate(all_mode_combos):
+        mode_a, mode_b = int(mode_a), int(mode_b)
+        if (freqs[mode_a].imag) > 1e-6 or (freqs[mode_b].imag) > 1e-6:
+            continue # pragma: no cover
+
+        displ_vec_a = displ_vecs[mode_a]
+        scaling_a = np.sqrt(HBAR / (2 * np.pi * freqs[mode_a] * 100 * C_LIGHT))
+
+        displ_vec_b = displ_vecs[mode_b]
+
+        scaling_b = np.sqrt(HBAR / (2 * np.pi * freqs[mode_b] * 100 * C_LIGHT))
+        for job_idx, [i, pt1, j, pt2] in enumerate(jobs_on_rank):
+
+            i, j = int(i), int(j)
+            positions = np.array(
+                init_geom + scaling_a * pt1 * displ_vec_a + scaling_b * pt2 * displ_vec_b
+            )
+            displ_mol = qml.qchem.Molecule(
+                molecule.symbols,
+                positions,
+                basis_name=molecule.basis_name,
+                charge=molecule.charge,
+                mult=molecule.mult,
+                unit="angstrom",
+                load_data=True,
+            )
+            displ_scf = single_point(displ_mol, method=method)
+            idx = mode_idx * len(jobs_on_rank) + job_idx
+            local_pes_twobody[idx] = (
+                displ_scf.e_tot - pes_onebody[mode_a, i] - pes_onebody[mode_b, j] - scf_result.e_tot
+            )
+
+            if do_dipole:
+                local_dipole_twobody[idx, :] = (
+                    get_dipole(displ_scf, method)
+                    - dipole_onebody[mode_a, i, :]
+                    - dipole_onebody[mode_b, j, :]
+                    - ref_dipole
+                )
+
+    if do_dipole:
+        return local_pes_twobody, local_dipole_twobody
+
+    return local_pes_twobody, None # pragma: no cover
```

Comment:
There shouldn't be a None here.

Reply:
A user can decide if they only want the PES or want both PES and dipole, if the dipole is not being calculated, it returns None for the dipole. Do you think we should handle this in some other manner?